### PR TITLE
Add some comments

### PR DESCRIPTION
**Pull request by TroyMitchell911**
## 拉取/合并请求描述：(PR description)

[
modified:   components/libc/posix/io/mman/mman.c           Add all function comments
modified:   components/libc/posix/io/mman/sys/mman.h    Add comments for all macros
modified:   components/libc/posix/io/poll/poll.c                   Add all function comments and comments to structure members
modified:   components/libc/posix/io/poll/poll.h                  Add all macro comments and comments to structure members
]

### 当前拉取/合并请求的状态 Intent for your PR

必须选择一项 Choose one (Mandatory):

- [ ] 本拉取/合并请求是一个草稿版本 This PR is for a code-review and is intended to get feedback
- [x] 本拉取/合并请求是一个成熟版本 This PR is mature, and ready to be integrated into the repo

### 代码质量 Code Quality：

我在这个拉取/合并请求中已经考虑了 As part of this pull request, I've considered the following:

- [x] 已经仔细查看过代码改动的对比 Already check the difference between PR and old code
- [x] 代码风格正确，包括缩进空格，命名及其他风格 Style guide is adhered to, including spacing, naming and other styles
- [x] 没有垃圾代码，代码尽量精简，不包含`#if 0`代码，不包含已经被注释了的代码 All redundant code is removed and cleaned up
- [x] 所有变更均有原因及合理的，并且不会影响到其他软件组件代码或BSP All modifications are justified and not affect other components or BSP
- [x] 对难懂代码均提供对应的注释 I've commented appropriately where code is tricky
- [x] 代码是高质量的 Code in this PR is of high quality
- [x] 已经使用[formatting](https://github.com/mysterywolf/formatting) 等源码格式化工具确保格式符合[RT-Thread代码规范](https://github.com/RT-Thread/rt-thread/blob/master/documentation/contribution_guide/coding_style_cn.md) This PR complies with [RT-Thread code specification](https://github.com/RT-Thread/rt-thread/blob/master/documentation/contribution_guide/coding_style_en.md)

### CODE REVIEW

File: components/libc/posix/io/poll/poll.h [hunk 1]
```diff
@@ -4,8 +4,9 @@
  * SPDX-License-Identifier: Apache-2.0
  *
  * Change Logs:
- * Date           Author       Notes
- * 2021-09-11     Meco Man     First version
+ * Date           Author            Notes
+ * 2021-09-11     Meco Man          First version
+ * 2024/03/29     TroyMitchelle     Add all macro comments and comments to structure members
```
Comment (mysterywolf) [inline components/libc/posix/io/poll/poll.h:9]:
```suggestion
 * 2024-03-29     TroyMitchelle     Add all macro comments and comments to structure members
```

File: components/libc/posix/io/poll/poll.h [hunk 2]
```diff
@@ -17,47 +18,48 @@ extern "C" {
...
-    short events;
-    short revents;
+    int fd;             /**< File descriptor. */
+    short events;       /**< Requested events. */
+    short revents;      /**< Returned events. */
 };
 #endif
 #endif /* !defined(POLLIN) && !defined(POLLOUT) */
 
+
```
Comment (mysterywolf) [inline components/libc/posix/io/poll/poll.h:62]:
```suggestion
```

File: components/libc/posix/io/poll/poll.c [hunk 3]
```diff
@@ -21,30 +22,45 @@
...
 
-struct rt_poll_node
-{
-    struct rt_wqueue_node wqn;
-    struct rt_poll_table *pt;
-    struct rt_poll_node *next;
+/**
+ * @brief   Structure representing a node in the poll table.
+ */
+struct rt_poll_node {
```
Comment (mysterywolf) [inline components/libc/posix/io/poll/poll.c:47]:
```suggestion
struct rt_poll_node
{
```

**Review by mysterywolf — CHANGES_REQUESTED** [review]

**Comment by mysterywolf** [comment]
你好，请修改以上问题，我是每个问题各列举了一个

**Comment by TroyMitchell911** [comment]
我明天将会修改

File: components/libc/posix/io/poll/poll.h (hunk 2 above)
Comment (mysterywolf) [reply components/libc/posix/io/poll/poll.h:62]: 这里多加了一个空行请删掉

File: components/libc/posix/io/mman/mman.c [hunk 4]
```diff
@@ -68,4 +85,4 @@ int munmap(void *addr, size_t length)
     }
 
     return -1;
-}
+}
```
Comment (mysterywolf) [inline components/libc/posix/io/mman/mman.c:88]: 这里不需要修改，请多按一个回车保持留出**一个**空行

**Review by mysterywolf — CHANGES_REQUESTED** [review]

**Review by mysterywolf — APPROVED** [review]

**Review by mysterywolf — APPROVED** [review]

**Pull request merged**